### ISSUE TITLE
Thermostat card

### ISSUE THREAD

**Issue by ciotlosm** [ciotlosm/custom-lovelace#4]
See https://codepen.io/dalhundal/pen/KpabZB  for inspiration for SVG

**Comment by iantrich** [comment]
I would like to see a more traditional view option as well (non-nest) that aligns better with the card-style in Lovelace. I would think we would want to encapsulate multiple styles within a single card or suite of cards to leverage common solutions such as handling of things like the state machine.

**Comment by ciotlosm** [comment]
I agree. Maybe we need to create a base model for service interaction and multiple styles. However I'd rather have at start this card implemented with all services and testing and then refactor to extract common things when building a second one. What do you think?

**Comment by ciotlosm** [comment]
Just an update: I've looked again at the implementation and I think it will require at least dual-temperature support before a first release. I'm in the process of refactoring it to allow this. Probably a usable version with service integration will be available by the end of the week.

**Comment by ciotlosm** [comment]
Testing version available: https://github.com/ciotlosm/custom-lovelace/tree/test_thermo
No controls yet and no dual temperature, just state.

**Comment by cicero200272** [comment]
Hi @ciotlosm, 

I just tested it with my Tado thermostats, works like a charm. Good work!

If Tado is switched off, it has "null" as target_temperature, the thermostat shows "0" (zero) in this case. Perhaps you could change that so no target temperature at all is shown?

Another special thing about Tado: there is no "away_mode" attribute at the climate entity, but the component defines an extra sensor "sensor.<entity_id>_tado_mode" which has either "HOME" or "AWAY" as state. Is it possible to also have a mapping here, probably even involving other entities or templating?

Cheers,
Tim.

**Comment by ciotlosm** [comment]
Added a TODO list here: https://github.com/ciotlosm/custom-lovelace/blob/test_thermo/thermostat-card/TODO.md

**Comment by alex-savin** [comment]
First of all, Thank you for you work! That is impressive! 

I have an issue with my thermostat card, is it possible to fix it
This is a data from my ZWave thermostat
```icon: mdi:thermostat-box
min_temp: 7
fan_mode: Auto Low
operation_list: auto,cool,off,heat
device_class: climate
value_index: 2
max_temp: 35
fan_list: Auto Low,On Low
node_id: XX
current_temperature: 22.5
fan_state: Running
supported_features: 193
value_instance: 1
operation_mode: cool
value_id: 7205XXXXXXXXX890
temperature: 21.7
operating_state: Cooling
unit_of_measurement: °C
friendly_name: Thermostat
```
![2018-07-28 12 42 49](https://user-images.githubusercontent.com/10366351/43358698-d10a3456-9263-11e8-9f48-581fbe6506a0.jpg)

**Comment by ciotlosm** [comment]
@alex-savin your data looks correct. Can you try incognito mode just to make sure it's not a caching issue? or append ?v=0.1 at the end of your resources url for the thermostat card

**Comment by alex-savin** [comment]
@ciotlosm adding `?v=0.1` fixed it!
Thank you very much!

**Comment by ciotlosm** [comment]
Ok, just released version 0.0.1 🎉
